packages/admin/src/table/filterbar/FilterBarActiveFilterBadge.tsx

```diff
@@ -0,0 +1,72 @@
+import { createStyles, Theme, Typography, WithStyles, withStyles } from "@material-ui/core";
+import * as React from "react";
+import { useForm } from "react-final-form";
+
+export type CometAdminFilterBarActiveFilterBadgeClassKeys = "hasValueCount";
+
+export const dirtyFieldsCount = (values: Record<string, any>, registeredFields: string[] = []) => {
+    let count = 0;
+    Object.entries(values).forEach(([fieldName, fieldValue]) => {
+        if (!fieldValue) {
+            return;
+        }
+
+        const isRegisteredField = registeredFields.includes(fieldName);
+        if (Array.isArray(fieldValue) && isRegisteredField) {
+            count += fieldValue.length;
+        } else if (typeof fieldValue === "object") {
+            if (Object.keys(fieldValue).length === 2 && "min" in fieldValue && "max" in fieldValue && isRegisteredField) {
+                count++;
+            } else {
+                count += dirtyFieldsCount(fieldValue, registeredFields);
+            }
+        } else if (isRegisteredField) {
+            count++;
+        }
+    });
+
+    return count;
+};
+
+const styles = (theme: Theme) =>
+    createStyles<CometAdminFilterBarActiveFilterBadgeClassKeys, FilterBarActiveFilterBadgeProps>({
+        hasValueCount: {
+            backgroundColor: `${theme.palette.grey[100]}`,
+            textAlign: "center",
+            borderRadius: "4px",
+            padding: "0 5px",
+            height: "20px",
+        },
+    });
+
+export interface FilterBarActiveFilterBadgeProps {
+    values: Record<string, any>;
+    calcNumberDirtyFields?: (values: Record<string, any>, registeredFields: string[]) => number;
+}
+
+export const FilterBarActiveFilterBadgeComponent: React.FC<WithStyles<typeof styles, true> & FilterBarActiveFilterBadgeProps> = ({
+    values,
+    calcNumberDirtyFields = dirtyFieldsCount,
+    classes,
+}) => {
+    const form = useForm();
+    const registeredFields = form.getRegisteredFields().map((field) => field.split(".").pop() as string);
```

Comment:
just using the last part isn't enough, that's not unique.

Reply:
does it need to be unique ? because i am just counting values. so if there are 2 keys with the same name, it still add 2 to the countValue :)

Reply:
yes, take the following example:
```
<FilterBarPopoverFilter label={"Test1"}>
    <Field name="test1.foo" type="text" component={FinalFormInput} fullWidth />
    <Field name="test1.bar" type="text" component={FinalFormInput} fullWidth />
</FilterBarPopoverFilter>
<FilterBarPopoverFilter label={"Test2"}>
    <Field name="test2.foo" type="text" component={FinalFormInput} fullWidth />
    <Field name="test2.bar" type="text" component={FinalFormInput} fullWidth />
</FilterBarPopoverFilter>
```

Reply:
both Badges will show 2 if you fill out all 4 FinalFormInput Fields :) because the registeredFields Method returns  ['test1.foo', 'test1.bar']  in the Test1 Form and  ['test2.foo', 'test2.bar'] in the Test2 Form .

Reply:
i just tested it and it really should be unique. i will try to fix that problem

Reply:
i found a solution by extending my recursive method :) i pushed my solution :)